### ISSUE TITLE
Feat. req. Direct link of nightly build (editor and exe) on main page

### ISSUE THREAD

**Issue by necros2k7** [adventuregamestudio/ags#1216]
subj

**Comment by ivan-mogilko** [comment]
There's a link, if you go to https://www.adventuregamestudio.co.uk/site/ags/ and click on "Other Downloads & Resources >>", it will open a list of links. There's "AGS Nightly Builds" among them.

**Comment by rofl0r** [comment]
which branch, @necros2k7 ? there's master aka ags3 legacy, ags3--sdl2 and ags4 (next).

**Comment by morganwillcock** [comment]
> I don't know if cirrus allows direct links to artifacts though, I was not able to get ones using their webpage?

You should be able to download them directly but artifacts will eventually expire so linking directly to them isn't a good idea. They would have to be uploaded somewhere else, but depending on the branch I don't think the concept of GitHub '[pre-]release' would apply.

> which branch, @necros2k7 ? there's master aka ags3 legacy, ags3--sdl2 and ags4 (next).

...and what is the 'main page' ?

**Comment by necros2k7** [comment]
By main I mean https://github.com/adventuregamestudio/ags
Yes ideally links should be to all latest ags3 legacy, ags3--sdl2 and ags4 (next) artifacts

**Comment by ivan-mogilko** [comment]
@necros2k7 there's a CI button which sais "builds | passing" in the readme, which you can see if you scroll down on that page.

Clicking on that button will direct you to the build server.

**Comment by necros2k7** [comment]
I thought Github Actions tab offers this functionality of setup and providing  latest builds

**Comment by ivan-mogilko** [comment]
Alright, so the idea is to have "nightly" builds (per branch?) available from some repository's "front page", and ensured to be persistent.
When another nightly is built, the link / download should be replaced.

Question is whether this is possible to do on Github, and what would be this "front page".
Frankly, I have no idea what "Github Actions" is, and whether it may allow this (I never used that tab).

**Comment by necros2k7** [comment]
> Alright, so the idea is to have "nightly" builds (per branch?) available from some repository's "front page", and ensured to be persistent. When another nightly is built, the link / download should be replaced.
> 
![zz](https://user-images.githubusercontent.com/16452453/229338140-1377cf55-64b7-4f70-8f21-f35640e8cdcc.JPG)

> Question is whether this is possible to do on Github, and what would be this "front page". Frankly, I have no idea what "Github Actions" is, and whether it may allow this (I never used that tab).

**Comment by ivan-mogilko** [comment]
Hmm, that's not at all the view that I'd expect from this... that's practically like embedding [our CI table](https://cirrus-ci.com/github/adventuregamestudio/ags) on github.

I thought the point is to have a simpler table, like a list of only the latest artifact links per branch with update date/time.

**Comment by morganwillcock** [comment]
GitHub Actions is a CI service, but not the one that AGS uses to build. It wouldn't be out the of the question to change it, but building with older versions of the build tools instead of tracking the latest versions makes it significantly more complicated to move. But regardless of which service is used, build artifacts will eventually expire because we do not download and store every build.

For releases and pre-releases which are created as GitHub Releases, those should be able to offered for download by implementing #616. That same page could also link to the latest CI downloads, although there is no guarantee that older builds are still there.

**Comment by ivan-mogilko** [comment]
> That same page could also link to the latest CI downloads, although there is no guarantee that older builds are still there.

What I wonder, is it theoretically possible to do this:
* detect when CI finishes building for branch X;
* download selected artifacts from CI server for the last task on branch X;
* upload these artifacts as attachments to github page.
* if there are already attachments for the branch X, delete these prior to uploading new ones.

**Comment by morganwillcock** [comment]
Theoretically, yes, although I'm not sure whether the GitHub Release metadata would make sense if the artifacts are being continually switched out. There is also no measure of how usable any of thoses builds are; potentially a working build gets replaced by a problematic one.

**Comment by ivan-mogilko** [comment]
> Theoretically, yes, although I'm not sure whether the GitHub Release metadata would make sense if the artifacts are being continually switched out. There is also no measure of how usable any of thoses builds are; potentially a working build gets replaced by a problematic one.

Ah, but I am not speaking of Release page, but some different kind of page. I don't know all available options there, so I am theorizing. For example, we have a manual uploaded to https://adventuregamestudio.github.io/ags-manual/. May we have another page with similar parent address, and host these links / attachments there?

**Comment by morganwillcock** [comment]
> For example, we have a manual uploaded to https://adventuregamestudio.github.io/ags-manual/

I don't think this hosting is intended for distributing binaries. There are transfer limits and excessive use would probably lead to the service being disabled. The files sizes wouldn't be large but someone could define a package that starts downloading builds from here.

> EDIT: hmm, maybe this is what you are speaking about https://github.com/adventuregamestudio/ags/issues/616#issuecomment-1493304407. (not sure)

What I describe there wouldn't host the binaries to make them available to download, it just provides links to where they are already hosted. The slightly fancier version uses the GitHub API to query GitHub Release metadata and build a more descriptive page, for example, the link to the latest release could list the version number instead of just saying "latest release". It could also list that latest pre-release because there is metadata that indicates the pre-release status of a release.

**Comment by ericoporto** [comment]
I have been thinking about this in the context of having whatever is the latest successful build so it can be used in a possible test game repository - see #2236 .

I had the following ideas to host "nightly" builds (I actually think it makes more sense to just be the latest successful binary from the master branch, and maybe ags4 branch too).

- create an arbitrary release in GitHub named TIP, UNTESTED, or whatever and just have the CI automatically push there for the respective branches, overwriting/removing whatever it was previously stored there.
- Push to [Cloudsmith](https://help.cloudsmith.io/docs/open-source-hosting-policy), but in the same sense directly from CI
- Push to some file hosting service like google drive (this is tricky because I can't figure it out if this is ok in their terms of service)
- Get it directly from CirrusCI, and then just write somewhere explaining how to get it from there or talk with CirrusCI if such feature would make sense

**Comment by ericoporto** [comment]
Made a question on the forums and got again the idea of a GitHub release that gets overwritten

https://github.com/orgs/community/discussions/77002

I updated the previous comment with additional alternatives.
